### PR TITLE
Disable Google Analytics when "do not track" is set (fixes #1204)

### LINKED ISSUE #1204: We should disable google analytics when "do not track" is set
Per Mozilla's [privacy policy](https://www.mozilla.org/en-US/privacy/websites/) for websites, we should disable google analytics when "do not track" is enabled.

We can determine whether do not track is enabled using this procedure:

https://dev.to/corbindavenport/how-to-correctly-check-for-do-not-track-with-javascript-135d

And then feed it into google analytics (when present) using this set of instructions:

https://developers.google.com/analytics/devguides/collection/gtagjs/user-opt-out

### PR DESCRIPTION
**Pull request by wlach**

### CODE REVIEW

**Comment by wlach** [comment]
Could one of you review/merge when you have time? This is a pretty easy one.

File: server/templates/ga.html [hunk 1]
```diff
@@ -1,8 +1,25 @@
...
-  gtag('js', new Date());
+  if (!window['ga-disable-{{ GA_TRACKING_ID }}']) {
+    window.dataLayer = window.dataLayer || [];
+    function gtag(){dataLayer.push(arguments);}
+    gtag('js', new Date());
 
-  gtag('config', '{{ GA_TRACKING_ID }}');
+    gtag('config', '{{ GA_TRACKING_ID }}');
+  } else {
+    console.log('Do not track enabled, google analytics disabled')
```
Comment (mdboom) [inline server/templates/ga.html]: Trying to do some basic local testing: I don't get this console message with "Do Not Track" turned on in my browser and hitting the local server.  But maybe GA never comes into play with the local server anyway?

Comment (wlach) [reply server/templates/ga.html]: Right, we don't include any google analytics code unless the GA_TRACKING_ID environment variable is set:

https://github.com/iodide-project/iodide/blob/1f9ac4b0e25299f76727e49e05c93c28de191482/server/context_processors.py#L9
https://github.com/iodide-project/iodide/blob/1f9ac4b0e25299f76727e49e05c93c28de191482/server/settings.py#L55

Comment (mdboom) [reply server/templates/ga.html]: Ok, in that case, :ship: it, and we'll come back around and test it on Heroku once it's deployed.

**Comment by wlach** [comment]
fwiw I tested it locally in my environment and it seemed to work as intended :)

**Pull request merged**